### DISCUSSION TITLE
ENVI Spectral Libraries

### DISCUSSION THREAD

**Discussion by grantboxer** [EnMAP-Box/enmap-box#299] [General]
Good morning,

I am trying to use the EnMAP and EnGEOMAP plug-ins for QGIS but I do not have access to ENVI spectral libraries. I have downloaded the USGS spectral libraries but they are not in a format that can be read by EnGeoMAP.

Any suggestions on where I can get the ENVI format spectral libraries?

Regards Grant Boxer

**Comment by janzandr** [comment]
Hi @grantboxer, please share the download link, or the USGS library itself.
Maybe we (most probably @jakimowb) will support that format directly in a future EnMAP-Box version.

**Comment by grantboxer** [comment]
Hi Andreas,

 

See this link “Spectral Library | U.S. Geological Survey (usgs.gov) <https://www.usgs.gov/labs/spectroscopy-lab/science/spectral-library> ”,

 

Cheers Grant

 

Grant Boxer

Consultant Geologist (FAIG R.P. Geo)

Perth, Western Australia

Email: ***@***.*** ***@***.***> 

Mobile: 0418 954 059

Skype: grantboxer

Website: https://grantboxer.github.io

**Comment by janzandr** [comment]
Alright, I found this 5GB zip file:
![image](https://user-images.githubusercontent.com/45030715/207836406-13b9cb2c-6b7e-4fbe-b683-9f7e77e439ab.png)
Is that correct?

**Comment by janzandr** [comment]
I had a look at the library and I think it can be imported quite easily. All spectra are indexed inside an HTML file, so we just have to parse the index and write everything into a spectral library format that the EnMAP-Box supports. I would use GeoJSON or GPKG.
![image](https://user-images.githubusercontent.com/45030715/207843823-e7d8dba5-63a2-471f-9c8d-269787be56a1.png)

@jakimowb, I could do that.

**Comment by grantboxer** [comment]
That would be great! Many thanks,

 

Cheers Grant

**Comment by jakimowb** [comment]
@grantboxer thanks for this idea. I opened #302 to add support for USGS Speclibs

@janzandr the 5Gb file seems to be a redundant collection of ASCII and SPECPR files (needs +2hour to get extracted on my system).

**Reply by janzandr** [reply]
Isn't that the USGS library? Is there a smaller version of it?

**Comment by grantboxer** [comment]
That’s great thanks

 

Grant

**Comment by grantboxer** [comment]
The zip file is a collection of all sorts of stuff. See this link for the ascii data “https://www.dropbox.com/scl/fo/nb2xylkzgdcnmyvhkj8hf/h?dl=0 <https://www.dropbox.com/scl/fo/nb2xylkzgdcnmyvhkj8hf/h?dl=0&rlkey=1y7kcstv0tgrqoyigxd12tdtq> &rlkey=1y7kcstv0tgrqoyigxd12tdtq:.

 

Not exactly sure which data is best for EnMAP. You might need to download and extract to see what is best.

 

Cheers Grant

**Comment by grantboxer** [comment]
Hi Andreas,

 

Just found this email again and reminded me about this.

 

Where did you get to with this, was it possible to put these into EnMAP / EnGeoMAP for sprectral mapping? Or perhaps a subset of key minerals for alteration mapping?

 

Cheers Grant

**Comment by janzandr** [comment]
> Where did you get to with this, was it possible to put these into EnMAP / EnGeoMAP for sprectral mapping? Or perhaps a subset of key minerals for alteration mapping?

Hi @grantboxer, we created an issue for the USGS Speclib import: #302

**Comment by janzandr** [comment]
An import algorithm will be available soon. See #302
